packages/spring/src/index.ts

```diff
@@ -0,0 +1,339 @@
+import { isServer } from "solid-js/web";
+
+// ===========================================================================
+// Internals
+// ===========================================================================
+
+// src/internal/client/types.d.ts
+
+type Task = {
+  abort(): void;
+  promise: Promise<void>;
+};
+
+type TickContext<T extends SpringTarget> = {
+  inv_mass: number;
+  dt: number;
+  opts: SpringOptions & { set: SpringSetter<T> };
+  settled: boolean;
+};
+
+type Raf = {
+  /** Alias for `requestAnimationFrame`, exposed in such a way that we can override in tests */
+  tick: (callback: (time: DOMHighResTimeStamp) => void) => any;
+  /** Alias for `performance.now()`, exposed in such a way that we can override in tests */
+  now: () => number;
+  /** A set of tasks that will run to completion, unless aborted */
+  tasks: Set<TaskEntry>;
+};
+
+type TaskCallback = (now: number) => boolean | void;
+
+type TaskEntry = { c: TaskCallback; f: () => void };
+
+// src/internal/client/timing.js
+
+/** SSR-safe RAF function. */
+const request_animation_frame = isServer ? () => {} : requestAnimationFrame;
+
+/** SSR-safe now getter. */
+const now = isServer ? () => Date.now() : () => performance.now();
```

Comment:
Isn't both `Date.now` and `performance.now` available in both node and browser?

Reply:
Yes. performance.now() will even get you higher resolution data on the server than on the client.

Reply:
Yup, just checked `performance.now()` and `Date.now` are both available in both node and browser. Just mostly copied this from the svelte source code. So I have no idea what should be preferred here:
https://github.com/sveltejs/svelte/blob/main/packages/svelte/src/internal/client/timing.js

In Svelte:
```
const now = BROWSER ? () => performance.now() : () => Date.now();
``` 

---

I'll take your advice on this one though, should I still use a ternary or just stick with one?

Reply:
Just pick one.
Will this code path even hit on the server? After it's inlined to createSpring? The whole primitive should be a normal signal on the server, no need to read current time then.

Reply:
So I tried `Date.now()` it doesn't work, it broke the spring. I'll stick with `performance.now()`.